kotlin/reflect/src/main/java/com/squareup/moshi/kotlin/reflect/KotlinJsonAdapter.kt

```diff
@@ -254,7 +250,9 @@ class KotlinJsonAdapterFactory : JsonAdapter.Factory {
 
     for (parameter in constructor.parameters) {
       val binding = bindingsByName.remove(parameter.name)
-      require(!(binding == null && !parameter.isOptional)) { "No property for required constructor $parameter" }
+      require(binding != null || parameter.isOptional) {
```

Comment:
`binding != null`? Is that right?

Reply:
Yeah this is just the flipped version of the `&&` before it. Autogenerated from intelliJ's inspection

Reply:
Ugh, I find the `require` difficult to parse since it's basically an else clause holding the condition, but my brain wants to read it as an `if`. That's gonna take some getting used to.